File: spring/src/main/java/com/example/kafkaproject/service/KafkaConsumerService.java
package com.example.kafkaproject.service;

import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Service;

@Service
public class KafkaConsumerService {

  @KafkaListener(topics = "${spring.kafka.topic.message-process-topic}", groupId = "spring-group")
  public void consume(String message) {
    System.out.println("Consumed message: " + message);
  }
}

File: kotlin/src/main/kotlin/com/example/kopring_kafka_consumer/service/KafkaConsumerService.kt
package com.example.kopring_kafka_consumer.service

import org.springframework.beans.factory.annotation.Value
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Service

@Service
class KafkaConsumerService {

    @KafkaListener(topics = ["\${spring.kafka.topic.message-process-topic}"], groupId = "my-group-id")
    fun consume(message: String) {
        println("Consumed message: $message")
    }
}
